### PR TITLE
Use "/" in $routes->resource and $routes->presenter to cause access failure.

### PR DESCRIPTION
**Pull request by vocaloidat**
In the official document: https://codeigniter.com/user_guide/incoming/restful.html
Tutorial:
$routes->resource('api/photo');
$routes->presenter('admin/photos');
Wrong execution guidance.
#### Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html

**Sorry, my English is not clear.**

can be seen:

```php
$routes->resource('api/photo');
$routes->presenter('admin/photos');
```

The code generated above is mapped (incorrect):

```php
$routes->put('api/photo/(:segment)','Api/Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin/Photos::update/$1');
```

But the standard should be (correct)

```php
$routes->put('api/photo/(:segment)','Api\Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin\Photos::update/$1');
```

Yes=‘\’, not=‘/’.
The command space USES = ‘\’, not=  ‘/’ as in the URL.

### So I added this code to prevent errors.

According to the current version, this sub-operation is wrong, and the URL link address (put, delete, etc.) cannot be accessed according to this. This submission request is required to complete the official complete update of the framework.

### CODE REVIEW

**Comment by MGatner** [comment]
I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

**Comment by vocaloidat** [comment]
> 我认为这些应该是路由，而不是名称空间。不使用手机时，我将不得不靠近看。

Yes, but it was executed according to the source code, and an error occurred. This modification can avoid mistakes.

**Comment by vocaloidat** [comment]
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

Yes, but it was executed according to the source code, and an error occurred. This modification can avoid mistakes.

**Comment by vocaloidat** [comment]
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html
can be seen:
$routes->resource('api/photo');
$routes->presenter('admin/photos');
According to the current version, this operation is wrong, and this submission request is required to complete the official complete update of the framework.

**Comment by vocaloidat** [comment]
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html
can be seen:
$routes->resource('api/photo');
$routes->presenter('admin/photos');
According to the current version, this sub-operation is wrong, and the URL link address (put, delete, etc.) cannot be accessed according to this. This submission request is required to complete the official complete update of the framework.

**Comment by vocaloidat** [comment]
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

#### Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html

can be seen:

```php
$routes->resource('api/photo');
$routes->presenter('admin/photos');
```

The code generated above is mapped (incorrect):

```php
$routes->put('api/photo/(:segment)','Api/Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin/Photos::update/$1');
```

But the standard should be (correct)

```php
$routes->put('api/photo/(:segment)','Api\Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin\Photos::update/$1');
```

Yes=‘\’, not=‘/’.

### So I added this code to prevent errors.



According to the current version, this sub-operation is wrong, and the URL link address (put, delete, etc.) cannot be accessed according to this. This submission request is required to complete the official complete update of the framework.

##### So I added this code to prevent errors.

**Comment by vocaloidat** [comment]
> I think these are supposed to be routes, not namespaces. I will have to look close when not on mobile.

#### Here, in the official statement, https://codeigniter.com/user_guide/incoming/restful.html
@MGatner 
can be seen:

```php
$routes->resource('api/photo');
$routes->presenter('admin/photos');
```

The code generated above is mapped (incorrect):

```php
$routes->put('api/photo/(:segment)','Api/Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin/Photos::update/$1');
```

But the standard should be (correct)

```php
$routes->put('api/photo/(:segment)','Api\Photo::update/$1');
....
$routes->post('admin/photos/(:segment)', 'Admin\Photos::update/$1');
```

Yes=‘\’, not=‘/’.
The command space USES = ‘\’, not= ‘/’ as in the URL.
### So I added this code to prevent errors.



According to the current version, this sub-operation is wrong, and the URL link address (put, delete, etc.) cannot be accessed according to this. This submission request is required to complete the official complete update of the framework.

##### So I added this code to prevent errors.

File: system/Router/RouteCollection.php [hunk 1]
```diff
@@ -807,7 +807,8 @@ public function resource(string $name, array $options = null): RouteCollectionIn
 		// resources are sent to, we need to have a new name
 		// to store the values in.
 		$new_name = ucfirst($name);
-
+		// replace / as \
+        $new_name = str_replace("/","\\",$new_name);
```
Comment (michalsn) [inline system/Router/RouteCollection.php:811]: Can we change it to one-liner which will `ucfirst` all parts of the namespace?
```php
$new_name = implode("\\", array_map('ucfirst', explode('/', $name)));
```

File: system/Router/RouteCollection.php [hunk 2]
```diff
@@ -921,7 +922,8 @@ public function presenter(string $name, array $options = null): RouteCollectionI
 		// resources are sent to, we need to have a new name
 		// to store the values in.
 		$newName = ucfirst($name);
-
+		// replace / as \
+        $newName = str_replace("/","\\",$newName);
```
Comment (michalsn) [inline system/Router/RouteCollection.php:926]: Same as above
```php
$newName = implode("\\", array_map('ucfirst', explode('/', $name)));
```

File: system/Router/RouteCollection.php (hunk 1 above)
Comment (vocaloidat) [reply system/Router/RouteCollection.php]: Nice, I'm sorry that due to regional reasons (China), there was a problem with VPN, which was not timely.

File: system/Router/RouteCollection.php [hunk 3]
```diff
@@ -806,8 +806,9 @@ public function resource(string $name, array $options = null): RouteCollectionIn
 		// In order to allow customization of the route the
 		// resources are sent to, we need to have a new name
 		// to store the values in.
-		$new_name = ucfirst($name);
-
+		//$new_name = ucfirst($name);
+		// Michal thought, modifying a single line in all parts of the namespace
```
Comment (michalsn) [inline system/Router/RouteCollection.php:810]: Please remove these two lines. Changes are tracked by git history so commenting the code is not necessary. Also, mentioning my name in the comment is not worth it :)

File: system/Router/RouteCollection.php [hunk 4]
```diff
@@ -920,8 +921,9 @@ public function presenter(string $name, array $options = null): RouteCollectionI
 		// In order to allow customization of the route the
 		// resources are sent to, we need to have a new name
 		// to store the values in.
-		$newName = ucfirst($name);
-
+		//$newName = ucfirst($name);
+		// Michal thought, modifying a single line in all parts of the namespace
```
Comment (michalsn) [inline system/Router/RouteCollection.php:925]: Same as above :)

**Comment by michalsn** [comment]
Thank you for your first contribution to this project!

**Pull request merged**